modules/core/src/commonMain/kotlin/fr/acinq/lightning/db/PaymentsDb.kt

```diff
@@ -465,103 +480,91 @@ data class LightningOutgoingPayment(
 
 sealed class OnChainOutgoingPayment : OutgoingPayment() {
     abstract override val id: UUID
-    abstract val miningFees: Satoshi
+    abstract val miningFee: Satoshi
     abstract val channelId: ByteVector32
     abstract val txId: TxId
+    abstract val liquidityPurchase: LiquidityAds.Purchase?
     abstract override val createdAt: Long
     abstract val confirmedAt: Long?
     abstract val lockedAt: Long?
     override val completedAt: Long? get() = lockedAt
     override val succeededAt: Long? get() = lockedAt
 
+    /** Fee paid to the Lightning Service Provider for this on-chain operation. */
+    val serviceFee: MilliSatoshi get() = liquidityPurchase?.fees?.serviceFee?.toMilliSatoshi() ?: 0.msat
+
+    /** If some liquidity was purchased, we paid a service fee on top of the mining fee. */
+    override val fees: MilliSatoshi get() = miningFee.toMilliSatoshi() + serviceFee
+
+    val liquidityPurchaseDetails: LiquidityAds.LiquidityTransactionDetails? get() = liquidityPurchase?.let { LiquidityAds.LiquidityTransactionDetails(txId, miningFee, it) }
+
     /** Helper method to facilitate updating child classes */
     fun setLocked(lockedAt: Long): OnChainOutgoingPayment =
         when (this) {
             is SpliceOutgoingPayment -> copy(lockedAt = lockedAt)
-            is SpliceCpfpOutgoingPayment -> copy(lockedAt = lockedAt)
-            is InboundLiquidityOutgoingPayment -> copy(lockedAt = lockedAt)
+            is SpliceCpfp -> copy(lockedAt = lockedAt)
+            is LiquidityPurchasePayment -> copy(lockedAt = lockedAt)
             is ChannelCloseOutgoingPayment -> copy(lockedAt = lockedAt)
         }
 
     /** Helper method to facilitate updating child classes */
     fun setConfirmed(confirmedAt: Long): OnChainOutgoingPayment =
         when (this) {
             is SpliceOutgoingPayment -> copy(confirmedAt = confirmedAt)
-            is SpliceCpfpOutgoingPayment -> copy(confirmedAt = confirmedAt)
-            is InboundLiquidityOutgoingPayment -> copy(confirmedAt = confirmedAt)
+            is SpliceCpfp -> copy(confirmedAt = confirmedAt)
+            is LiquidityPurchasePayment -> copy(confirmedAt = confirmedAt)
             is ChannelCloseOutgoingPayment -> copy(confirmedAt = confirmedAt)
         }
 }
 
+/** A splice transaction that sends a payment to an on-chain address. */
 data class SpliceOutgoingPayment(
     override val id: UUID,
     val recipientAmount: Satoshi,
     val address: String,
-    override val miningFees: Satoshi,
+    override val miningFee: Satoshi,
     override val channelId: ByteVector32,
     override val txId: TxId,
+    override val liquidityPurchase: LiquidityAds.Purchase?,
```

Comment:
AFAICT this parameter is not used yet. It should be readonly to `null`, like in `SpliceCpfp`.